x-pack/plugins/apm/public/components/app/service_details/service_icons/index.test.tsx

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { fireEvent, render } from '@testing-library/react';
+import { CoreStart } from 'kibana/public';
+import { merge } from 'lodash';
+// import { renderWithTheme } from '../../../../utils/testHelpers';
+import React, { ReactNode } from 'react';
+import { createKibanaReactContext } from 'src/plugins/kibana_react/public';
+import { MockUrlParamsContextProvider } from '../../../../context/url_params_context/mock_url_params_context_provider';
+import { ApmPluginContextValue } from '../../../../context/apm_plugin/apm_plugin_context';
+import {
+  mockApmPluginContextValue,
+  MockApmPluginContextWrapper,
+} from '../../../../context/apm_plugin/mock_apm_plugin_context';
+import * as fetcherHook from '../../../../hooks/use_fetcher';
+import { ServiceIcons } from './';
+
+const KibanaReactContext = createKibanaReactContext({
+  usageCollection: { reportUiCounter: () => {} },
+} as Partial<CoreStart>);
+
+const addWarning = jest.fn();
+const httpGet = jest.fn();
+
+function Wrapper({ children }: { children?: ReactNode }) {
+  const mockPluginContext = (merge({}, mockApmPluginContextValue, {
+    core: { http: { get: httpGet }, notifications: { toasts: { addWarning } } },
+  }) as unknown) as ApmPluginContextValue;
+
+  return (
+    <KibanaReactContext.Provider>
+      <MockApmPluginContextWrapper value={mockPluginContext}>
+        <MockUrlParamsContextProvider
+          params={{
+            rangeFrom: 'now-15m',
+            rangeTo: 'now',
+            start: 'mystart',
+            end: 'myend',
+          }}
+        >
+          {children}
+        </MockUrlParamsContextProvider>
+      </MockApmPluginContextWrapper>
+    </KibanaReactContext.Provider>
+  );
+}
+
+describe('ServiceIcons', () => {
+  describe('icons', () => {
+    it('Shows loading spinner while fetching data', () => {
+      jest.spyOn(fetcherHook, 'useFetcher').mockReturnValue({
+        data: undefined,
+        status: fetcherHook.FETCH_STATUS.LOADING,
+        refetch: jest.fn(),
+      });
+      const { getByTestId, queryAllByTestId } = render(
+        <Wrapper>
+          <ServiceIcons serviceName="foo" />
+        </Wrapper>
+      );
+      expect(getByTestId('loading')).toBeInTheDocument();
+      expect(queryAllByTestId('java').length).toEqual(0);
```

Comment:
You can use `toHaveLength(0)`; it's a bit shorter.

I'd prefer not having so many expectations in one test.

Reply:
Thanks, I didn't know about it, it looks way better now.